internal/consensus/reactor.go

```diff
@@ -28,36 +28,36 @@ var (
 
 // GetChannelDescriptor produces an instance of a descriptor for this
 // package's required channels.
-func GetChannelDescriptors() []*p2p.ChannelDescriptor {
-	return []*p2p.ChannelDescriptor{
-		{
+func getChannelDescriptors() map[p2p.ChannelID]*p2p.ChannelDescriptor {
+	return map[p2p.ChannelID]*p2p.ChannelDescriptor{
```

Comment:
I think a `ChannelDescriptorSet` struct, with each of these named as fields is a bit preferable to using a map.

Reply:
this makes it harder, I think, to implement the shim in the testing, because you can't do dot-access via a variable. 

the worst thing that happens is we add a channel, forget to add it everywhere, and the reactor panics on construction, which I think is actually a pretty good possible outcome.